From cbd7d45aafb9c17d8bbcc6a6e041c7889f319222 Mon Sep 17 00:00:00 2001
From: Peter Vary <pvary@cloudera.com>
Date: Thu, 27 Apr 2017 10:49:45 +0200
Subject: [PATCH 1084/1164] CDH-52465: HIVE-16345: BeeLineDriver should be
 able to run qtest files which are using default
 database tables (Peter Vary via Yongzhi Chen)    
        Additional CDH specific fix

Change-Id: Ia024cca70ba77e17cec21f96507db184e6878e37
---
 data/conf/hive-site.xml                            |   12 ++++++++++++
 .../hive/beeline/qfile/QFileBeeLineClient.java     |    3 ++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/data/conf/hive-site.xml b/data/conf/hive-site.xml
index 0721ccb..437daa4 100644
--- a/data/conf/hive-site.xml
+++ b/data/conf/hive-site.xml
@@ -64,6 +64,12 @@
 </property>
 
 <property>
+  <name>hive.stats.dbconnectionstring</name>
+  <value>jdbc:derby:;databaseName=${test.tmp.dir}/TempStatsStore;create=true</value>
+</property>
+
+
+<property>
   <name>javax.jdo.option.ConnectionDriverName</name>
   <value>org.apache.derby.jdbc.EmbeddedDriver</value>
 </property>
@@ -200,6 +206,12 @@
 </property>
 
 <property>
+  <name>hive.stats.dbclass</name>
+  <value>jdbc:derby</value>
+  <description>The storage for temporary stats generated by tasks. Currently, jdbc, hbase and counter types are supported</description>
+</property>
+
+<property>
   <name>hive.stats.key.prefix.reserve.length</name>
   <value>0</value>
 </property>
diff --git a/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFileBeeLineClient.java b/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFileBeeLineClient.java
index 7244bf8..3e6475c 100644
--- a/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFileBeeLineClient.java
+++ b/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFileBeeLineClient.java
@@ -46,7 +46,8 @@ protected QFileBeeLineClient(String jdbcUrl, String jdbcDriver, String username,
           "!set showelapsedtime false",
           "!set trimscripts false",
           "!set maxwidth -1",
-          "!connect " + jdbcUrl + " " + username + " " + password + " " + jdbcDriver
+          "!connect " + jdbcUrl + " " + username + " " + password + " " + jdbcDriver,
+          "set hive.stats.dbclass=fs"
         });
   }
 
-- 
1.7.9.5

